src/Types.hs

```diff
@@ -27,38 +27,63 @@ import qualified Data.Csv as CSV
 
 -- | Globally Unique ID (for reference in the database).  (FIXME: should we have different id types
 -- for different object types?)
-type GUID = Integer
+newtype GUID a = GUID Integer
+  deriving (Eq, Ord, Show, Read, Generic)
 
-data MetaInfo = MetaInfo
-    { _metaId        :: GUID
-    , _metaCreatedBy :: GUID
+data MetaInfo a = MetaInfo
+    { _metaId        :: GUID a
+    , _metaCreatedBy :: GUID User
     , _metaCreatedAt :: Timestamp
-    , _metaChangedBy :: GUID
+    , _metaChangedBy :: GUID User
     , _metaChangedAt :: Timestamp
     }
   deriving (Eq, Ord, Show, Read, Generic)
 
-newtype Article = Article { fromArticle :: [ST] }
+-- | Markdown content.
+newtype Document = Document { fromDocument :: ST }
```

Comment:
Agreed.

(@andorp, you had a nice hack yesterday for rendering markdown in the browser.  Could you paste that here for later reference?  Or even somewhere appropriate in the code?  Thanks!)